src/scripts/generate.ts

```diff
@@ -67,40 +73,126 @@ const generateAndExportTokens = async (
   teamId: FigmaTeamId,
   fileKeyFilters?: FigmaFileKey[]
-): Promise<Token[]> => {
+) => {
   info("Starting team styles update...");
-  const teamStyleTokens = await fetchAndFormatTeamStyles(
+  const tokens = await fetchAndFormatTeamStyles(
     figmaApiClient,
     teamId,
     fileKeyFilters
   );
-  const rootTokenCollection = groupTokens(teamStyleTokens, logger);
+  const rootTokenCollection = groupTokens(tokens, logger);
   const tokenValues = unwrapTokenValues(rootTokenCollection);
-  await generateExportedTS(rootTokenCollection, tokenValues, teamStyleTokens);
-  return teamStyleTokens;
+  return {
+    tokens,
+    rootTokenCollection,
+    tokenValues,
+  };
+};
+
+/**
+ * Fetches icons metadata from Figma API and filters them based on the icon file key.
+ *
+ * @param {FigmaApiClient} figmaApiClient - the Figma API client
+ * @param {FigmaTeamId} teamId - the ID of the Figma team
+ * @param {FigmaFileKey} iconFileKey - the key of the Figma file containing icons
+ * @return {Promise<Component[]>} an array of components representing the icons metadata
+ */
+const fetchIconsMetadata = async (
+  figmaApiClient: FigmaApiClient,
+  teamId: FigmaTeamId,
+  iconFileKey: FigmaFileKey
+) => {
+  info("Fetching all components in the exported team styles...");
+  const componentsResponse = await figmaApiClient.getTeamComponents(teamId, {
+    page_size: 10000,
+  });
+
+  if (!componentsResponse) return [];
+  if (componentsResponse.error) {
+    warn(componentsResponse.error);
+    return [];
+  }
+  if (componentsResponse.meta.components.length === 0) {
+    return [];
+  }
+  info(
+    `Filtering ${componentsResponse.meta.components.length} components based on the icon file key...`
+  );
+  const filteredComponents = componentsResponse.meta.components.filter(
+    (item) => item.file_key === iconFileKey
+  );
+  log(filteredComponents);
+  return filteredComponents;
 };
 
 /**
- * Async function that initialized the Figma api client, fetches up-to-date Figma files and returns generated tokens.
- * @returns {Promise<Token[]>} - Array containing all generated tokens according to env config
+ * Generate and export icons from Figma.
+ *
+ * @param {FigmaApiClient} figmaApiClient - The Figma API client
+ * @param {FigmaTeamId} teamId - The team ID in Figma
+ * @param {FigmaFileKey[]} iconFileKey - The key of the Figma file containing the icons
+ * @return {Promise<Icon[]>} A promise that resolves to an array of icons
  */
-export const generate = async (): Promise<Token[]> => {
+const generateAndExportIcons = async (
+  figmaApiClient: FigmaApiClient,
+  teamId: FigmaTeamId,
+  [iconFileKey]: FigmaFileKey[]
+): Promise<Icon[]> => {
+  info("Starting icons update...");
+  const iconComponentsMetadata = await fetchIconsMetadata(
+    figmaApiClient,
+    teamId,
+    iconFileKey
+  );
+  const iconsSvg = await fetchIconSvgs(
+    figmaApiClient,
+    iconComponentsMetadata,
+    iconFileKey,
+    logger
+  );
+  return iconsSvg;
```

Comment:
Pourquoi tu as créer une const iconsSvg ? 
Autant faire ça: 
```suggestion
  return await fetchIconSvgs(
    figmaApiClient,
    iconComponentsMetadata,
    iconFileKey,
    logger
  );;
```

Reply:
j'avais besoin de la log aha